File: docs/components_page/components/modal/centered.R
library(dashBootstrapComponents)
library(dashHtmlComponents)

modal <- htmlDiv(
  list(
    dbcButton("Open", id = "open-centered"),
    dbcModal(
      list(
        dbcModalHeader(dbcModalTitle("Header"), close_button = TRUE),
        dbcModalBody("This modal is vertically centered"),
        dbcModalFooter(
          dbcButton(
            "Close",
            id = "close-centered", className = "ms-auto", n_clicks = 0
          )
        )
      ),
      id = "modal-centered",
      is_open = FALSE,
      centered = TRUE
    )
  )
)


app$callback(
  output("modal-centered", "is_open"),
  list(
    input("open-centered", "n_clicks"),
    input("close-centered", "n_clicks"),
    state("modal-centered", "is_open")
  ),
  function(n1, n2, is_open) {
    if (n1 > 0 | n2 > 0) {
      return(!is_open)
    }
    return(is_open)
  }
)

File: docs/components_page/components/modal/centered.jl
using DashBootstrapComponents

modal = html_div([
    dbc_button("Open", id = "open-centered"),
    dbc_modal(
        [
            dbc_modalheader(dbc_modaltitle("Header"), close_button = true),
            dbc_modalbody("This modal is vertically centered"),
            dbc_modalfooter(
                dbc_button(
                    "Close",
                    id = "close-centered",
                    className = "ms-auto",
                    n_clicks = 0,
                ),
            ),
        ],
        id = "modal-centered",
        centered = true,
        is_open = false,
    ),
]);

callback!(
    app,
    Output("modal-centered", "is_open"),
    Input("open-centered", "n_clicks"),
    Input("close-centered", "n_clicks"),
    State("modal-centered", "is_open"),
) do n1, n2, is_open
    return n1 > 0 || n2 > 0 ? is_open == 0 : is_open
end;
